src/gc.js

```diff
@@ -375,15 +419,8 @@ function h$follow(obj, sp) {
                     for(var i=0;i<s.length;i++) ADDW(s[i]);
                 }
             } else if(c instanceof h$Weak) {
-                TRACE_GC("marking weak reference");
-                if(c.keym.m === mark) {
-                    if(c.val !== null && !IS_MARKED(c.val)) ADDW(c.val);
-                } else {
```

Comment:
I removed the `ADDW()` here, without realy understanding what it does... Maybe not an good idea?